src/main/java/org/tron/common/runtime/vm/VM.java

```diff
@@ -27,6 +29,9 @@
     // used to reduce expensive BigInt arithmetic
     private static BigInteger MAX_MEM_SIZE = BigInteger.valueOf(Integer.MAX_VALUE);
 
+  // 3MB
+  private static BigInteger MEM_LIMIT = BigInteger.valueOf(3 * 1024 * 1024);
```

Comment:
![Bug](https://sonarsource.github.io/sonar-github/bug.png 'Bug') Bug: Cast one of the operands of this multiplication operation to a "long". ([squid:S2184](https://sonarcloud.io/organizations/tron-zhaohong/rules#rule_key=squid%3AS2184))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=java-tron&pullRequest=1191&issues=AWTgaItsrFWWQMKkhQvK&open=AWTgaItsrFWWQMKkhQvK)

[ISSUE_KEY]: AWTgaItsrFWWQMKkhQvK